### ISSUE TITLE
[Question] Logs require an axe by default

### ISSUE THREAD

**Issue by NillerMedDild** [tyra314/ToolProgression#7]
Hey! It seems that all wood logs require an axe to break, by default. Is this intended?

Minecraft 1.12.2
Forge 1.12.2 - 14.23.0.2512
toolprogression-1.12-1.2.9

**Comment by tyra314** [comment]
By default, the mod shouldn't change any behavior. To configure the mod is completly up to the pack dev. So yes, this is a bug.

**Comment by tyra314** [comment]
Having said that, it doesn't happen to me. Neither in my 1.12 dev environment, nor in a fresh 1.12.2 instance. Could you check your config?

**Comment by NillerMedDild** [comment]
This is my entire block_overwrites.cfg - That should be the only file that could do this, right?

https://pastebin.com/jpUpktH5

**Comment by tyra314** [comment]
Okay, I see the problem. It's probably due to the lack of documentation, shame on me. For my defense, it sucks writing documenation and most documentation doesn't help users anyway.

However, having these entries:

```
    # minecraft:log2:0 [default: Axe=-1]
    S:"minecraft:log2:0"=Axe=-1
```

in your block_overwrite.cfg does in fact result into "harvesting logs requires **any** axe".

**TL;DR If you don't want changes, keep your overwrite configs empty, as even copying the default entry possibly changes the behavior.**

The thing is, the determination of harvestability is more complex than simply comparing harvest levels. (More details [here](https://github.com/tyra314/ToolProgression/wiki/Concepts)). As mentioned on the wiki page, there is a concept called "Block.Material", which determine if a tool is required or optional for harvesting a given block. For instance, the material of minecraft:log defines tools to be optional, but the material of minecraft:stone defines tools to be required.
However, I have to skip some Vanilla/Forge code paths and determine the harvestability on my own. In this process, I forego materials, because different block (types) share the same Block.Material. This means that any block overwrite specified effectively also set the tool to be required. Thus, copying the default entry into the overwrite config may or may not change the default behavior. For instance, copying the default of minecraft:stone doesn't make any difference, because it already requires a pickaxe. But in case of logs, it makes a difference, as you experienced first hand. This whole behavior exposes one caveat of my mod: **You can't change the effective tool of a block.**

**Comment by NillerMedDild** [comment]
Ehm... The issue occured with an empty block_overwrites.cfg, the entries are just there to remedy the issue.

**Comment by NillerMedDild** [comment]
I found the issue though - It's a conflict between Harvest Tweaks and Tool Progression that causes it, with an empty block_overwrites.cfg - My bad for using them both though. Thank you for taking your time to help me out regardless :D

**Comment by tyra314** [comment]
Oh, I see, thanks for letting me know.
